### PR TITLE
feat(dashboards): Add equations to Big Number Widget (Backend API change)

### PR DESCRIPTION
**Pull request by harshithadurai**
https://github.com/getsentry/sentry/issues/68167

Adding the API changes with respect to the frontend changes to add equations to Big Number: https://github.com/getsentry/sentry/pull/77790

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py
Comment (narsaynorath) [inline tests/sentry/api/endpoints/test_organization_dashboard_details.py:323]: Can we also port these tests over to `test_organization_dashboards.py`?

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py
Comment (narsaynorath) [inline tests/sentry/api/endpoints/test_organization_dashboard_details.py:615]: So this doesn't properly validate that `data["widgets"][0]["queries"][0] == queries[0]` completely, it's a helper that checks for specific fields so right now it doesn't check that `selectedAggregate` is there. We should update `assert_serialized_widget_query` to check for the `selectedAggregate` field and validate that it's set

**Review by narsaynorath — APPROVED** [review]
Just left some comments about the test but it's looking good! I'm going to approve and assume you'll address the comments. We still need to get a review from a migrations owner before merging

File: src/sentry/api/serializers/models/dashboard.py
Comment (narsaynorath) [inline src/sentry/api/serializers/models/dashboard.py:44]: Actually I just realized we need to update the API docs for this change. Since `selectedAggregate` may not always be set, let's just redefine this so it can be `null` on the frontend. You'll want to change it to `Optional[int]`, check out the code snippet here: https://develop.sentry.dev/application/api/public/#success-responses. Optional needs to be imported like `from typing import Optional`

You'll also need to update the API examples in `src/sentry/apidocs/examples/dashboard_examples.py` so the widget queries have `selectedAggregate` set. It's fine if we just set them to `None`.

**Review by markstory — APPROVED** [review]
Schema changes look good.

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py
Comment (harshithadurai) [reply tests/sentry/api/endpoints/test_organization_dashboard_details.py:323]: moved the dashboard level test (test_dashboard_widget_query_returns_selected_aggregate)

**Review by narsaynorath — APPROVED** [review]
:shipit:

**Pull request merged**
